AudioFile.cpp

```diff
@@ -351,12 +351,18 @@ bool AudioFile<T>::decodeWaveFile (std::vector<uint8_t>& fileData)
                 int32_t sampleAsInt = 0;
                 sampleAsInt = (fileData[sampleIndex + 2] << 16) | (fileData[sampleIndex + 1] << 8) | fileData[sampleIndex];
                 
-                if (sampleAsInt & 0x800000) //  if the 24th bit is set, this is a negative number in 24-bit world
+                if (sampleAsInt & (1 << 23)) //  if the 24th bit is set, this is a negative number in 24-bit world
                     sampleAsInt = sampleAsInt | ~0xFFFFFF; // so make sure sign is extended to the 32 bit float
 
-                T sample = (T)sampleAsInt / (T)8388608.;
+                T sample = (T)sampleAsInt / (T)(1 << 23);
                 samples[channel].push_back (sample);
             }
+            else if (bitDepth == 32)
+            {
+              int32_t sampleAsInt = fourBytesToInt (fileData, sampleIndex);
+              T sample = (T)sampleAsInt / (T)(INT32_MAX);
+              samples[channel].push_back (sample);
+            }
```

Comment:
As per your input in the other comment, I changed INT_MAX to INT32_MAXin the above. It makes a lot of sense to be 32-bit specific here.

Reply:
I think we might be better here to use `std::numeric_limits<std::int32_t>::max()` as it is a bit more modern. Or to hardcode the value as it doesn't change. Depends how many people would try to use the library on older platforms (I'm pretty sure I've got some other C++11 stuff in here already though).

Reply:
Personally, I hate "magic numbers", even if they don't change..  I fully agree on `std::numeric_limits<std::int32_t>::max()` - that is indeed the modern way and it conveys what the number is supposed to be 👍 

I think C++11 is a good baseline standard to aim for this project. I tried compiling for C++98 as a quick test, and got like 20 errors, mostly due to closing template brackets needing a space (`> >`).

Reply:
Ok, excellent. I've marked one other issue above unresolved (the `1 << 31` thing) and then I can merge this in and  have a look :)